files/image_config/caclmgrd/caclmgrd

```diff
@@ -358,64 +406,88 @@ class ControlPlaneAclManager(object):
         Config DB, translates control plane ACLs into a list of iptables
         commands and runs them.
         """
-        iptables_cmds = self.get_acl_rules_and_translate_to_iptables_commands()
-
+        iptables_cmds = self.get_acl_rules_and_translate_to_iptables_commands(namespace)
         log_info("Issuing the following iptables commands:")
         for cmd in iptables_cmds:
             log_info("  " + cmd)
 
         self.run_commands(iptables_cmds)
 
-    def notification_handler(self, key, data):
-        log_info("ACL configuration changed. Updating iptables rules for control plane ACLs...")
-        self.update_control_plane_acls()
-
     def run(self):
-        # Unconditionally update control plane ACLs once at start
-        self.update_control_plane_acls()
-
-        # Subscribe to notifications when ACL tables or rules change
-        self.config_db.subscribe(self.ACL_TABLE,
-                lambda table, key, data: self.notification_handler(key, data))
-        self.config_db.subscribe(self.ACL_RULE,
-                lambda table, key, data: self.notification_handler(key, data))
-
-        # Indefinitely listen for Config DB notifications
-        self.config_db.listen()
-
+        # Select Time-out for 10 Seconds
+        SELECT_TIMEOUT_MS = 1000 * 10
+
+        # Initlaize Global config that loads all database*.json
+        if device_info.is_multi_npu():
+            swsscommon.SonicDBConfig.initializeGlobalConfig()
+        # Create the Select object
+        sel = swsscommon.Select()
+        # Map of Namespace <--> susbcriber table's object
+        config_db_subscriber_table_map = {}
+        # Loop through all asic namespaces (if present) and host (namespace='')
+        for namespace in self.config_db_map.keys():
+            # Unconditionally update control plane ACLs once at start on given namespace
+            self.update_control_plane_acls(namespace)
+            # Connect to Config DB of given namespace
+            acl_db_connector = swsscommon.DBConnector("CONFIG_DB", 0, False, namespace)
+            # Subscribe to notifications when ACL tables changes
+            subscribe_acl_table = swsscommon.SubscriberStateTable(acl_db_connector, swsscommon.CFG_ACL_TABLE_TABLE_NAME)
+            # Subscribe to notifications when ACL rule tables changes
+            subscribe_acl_rule_table = swsscommon.SubscriberStateTable(acl_db_connector, swsscommon.CFG_ACL_RULE_TABLE_NAME)
+            # Add both tables to the selectable object
+            sel.addSelectable(subscribe_acl_table)
+            sel.addSelectable(subscribe_acl_rule_table)
+            # Update the map
+            config_db_subscriber_table_map[namespace] = []
+            config_db_subscriber_table_map[namespace].append(subscribe_acl_table)
+            config_db_subscriber_table_map[namespace].append(subscribe_acl_rule_table)
+        # Loop on select to see if any event happen on config db of any namespace
```

Comment:
Please add a newline above this comment to visually break things up

Reply:
@jleveque Done